src/WebJobs.Extensions.ApiHub/ApiHubFile.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Azure.ApiHub;
+using Microsoft.Azure.WebJobs.Extensions.ApiHub.Common;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ApiHub
+{
+    class ApiHubFile : IFileStreamProvider
+    {
+        private readonly IFileItem _fileSource;
+
+        public string Path
+        {
+            get
+            {
+                return _fileSource.Path;
+            }
+        }
+
+        public ApiHubFile(IFolderItem rootFolder, string path)
+        {
+            _fileSource = rootFolder.CreateFileAsync(path, true).GetAwaiter().GetResult();
```

Comment:
@MikeStall @safihamid discussed this in the ApiHub SDK. CreateFile is returning the structure needed to create the file. Some of the SAAS providers dont allow creating empty file. So to make the SDK friendly CreateFile() is returning a "pointer" but the actual file is created on first write.